lib/AddonSignals.py

```diff
@@ -22,102 +20,127 @@ def _perf_clock():
 
     def registerSlot(self, signaler_id, signal, callback):
+        """Register a slot in the AddonSignals receiver"""
         if signaler_id not in self._slots:
             self._slots[signaler_id] = {}
         self._slots[signaler_id][signal] = callback
 
     def unRegisterSlot(self, signaler_id, signal):
+        """Unregister a slot in the AddonSignals receiver"""
         if signaler_id not in self._slots:
             return
         if signal not in self._slots[signaler_id]:
             return
         del self._slots[signaler_id][signal]
 
     def onNotification(self, sender, method, data):
-        if not sender[-7:] == '.SIGNAL':
+        """The Kodi Monitor event handler for notifications"""
+        if not sender.endswith('.SIGNAL'):
             return
         sender = sender[:-7]
         if sender not in self._slots:
             return
         signal = method.split('.', 1)[-1]
         if signal not in self._slots[sender]:
             return
-        self._slots[sender][signal](_decodeData(data))
+        self._slots[sender][signal](_decode_data(data))
 
 
-class CallHandler(object):
-    #pylint: disable=too-many-arguments
-    def __init__(self, signal, data, source_id, timeout=1000, use_timeout_exception=False):
+class CallHandler(object):  # pylint: disable=bad-option-value,useless-object-inheritance
+    """The AddonSignals event handler class"""
+    def __init__(self, signal, data, source_id, timeout=1000, use_timeout_exception=False):  # pylint: disable=too-many-arguments
+        """The CallHandler constructor"""
         self.signal = signal
         self.data = data
         self.timeout = timeout
-        self.sourceID = source_id
+        self.source_id = source_id
         self._return = None
         self.is_callback_received = False
         self.use_timeout_exception = use_timeout_exception
-        registerSlot(self.sourceID, '_return.{0}'.format(self.signal), self.callback)
-        sendSignal(signal, data, self.sourceID)
+        registerSlot(self.source_id, '_return_{0}'.format(self.signal), self.callback)
+        sendSignal(signal, data, self.source_id)
 
     def callback(self, data):
+        """Method to register function as callback"""
         self._return = data
         self.is_callback_received = True
 
     def waitForReturn(self):
-        monitor = xbmc.Monitor()
+        """Wait for callback to trigger"""
+        monitor = Monitor()
         end_time = _perf_clock() + (self.timeout / 1000)
         while not self.is_callback_received:
             if _perf_clock() > end_time:
                 if self.use_timeout_exception:
-                    unRegisterSlot(self.sourceID, self.signal)
+                    unRegisterSlot(self.source_id, self.signal)
                     raise WaitTimeoutError
                 break
-            elif monitor.abortRequested():
+            if monitor.abortRequested():
```

Comment:
try look here this was "elif"

Reply:
Sure, but pylint complains because it is actually not needed, the previous always breaks.